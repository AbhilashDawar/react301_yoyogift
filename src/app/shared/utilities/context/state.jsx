import * as React from 'react';

let ContextState = React.createContext();

let initialState = {
    showLoader: false,
    // filterParams: {
    //     startValue: config.rangeDefaults.min,
    //     endValue: config.rangeDefaults.max,
    //     discountOption: 0
    // }
};

let reducer = (state, action) => {
    switch (action.type) {
        case "RESET":
            return initialState;
        case "SHOW":
            return { ...state, showLoader: true };
        case "HIDE":
            return { ...state, showLoader: false };
        // case "UPDATE_FILTERS":
        //     return { ...state, filterParams: action.payload };
        default:
            return { ...state };
    }
};

function ContextStateProvider(props) {
    let [state, dispatch] = React.useReducer(reducer, initialState);
    let value = { state, dispatch };

    return (
        <ContextState.Provider value={value}>{props.children}</ContextState.Provider>
    );
}

let ContextStateConsumer = ContextState.Consumer;

export { ContextState, ContextStateProvider, ContextStateConsumer };