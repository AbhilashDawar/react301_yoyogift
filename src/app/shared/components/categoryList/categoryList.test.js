import React from 'react';
import CategoryList from './index.jsx';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { ContextStateProvider } from '../../utilities/context/state.jsx';

test('CategoryList Component', () => {
    const tree = renderer.create(
        <MemoryRouter>
            <ContextStateProvider>
                <CategoryList />
            </ContextStateProvider>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});