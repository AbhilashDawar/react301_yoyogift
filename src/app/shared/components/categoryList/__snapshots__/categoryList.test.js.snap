// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList Component 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 makeStyles-categoryPanel-2 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 MuiGrid-align-items-xs-flex-end MuiGrid-align-content-xs-flex-end MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <p
        className="MuiTypography-root makeStyles-bold-3 MuiTypography-body1"
      >
        Categories:
      </p>
    </div>
  </div>
</div>
`;
